castai/resource_castai_cluster.go

```diff
@@ -172,9 +309,54 @@ func resourceCastaiClusterRead(ctx context.Context, data *schema.ResourceData, m
 		data.Set(ClusterFieldKubeconfig, []interface{}{})
 	}
 
+	policies, err := client.GetPoliciesWithResponse(ctx,sdk.ClusterId(data.Id()))
+	if checkErr := sdk.CheckGetResponse(policies, err); checkErr == nil {
+		if err := data.Set(PolicyFieldAutoscalerPolicies, flattenAutoscalerPolicies(policies.JSON200)); err != nil {
+			return nil
+		}
+	} else {
+		log.Printf("[WARN] autoscaling policies are not available for cluster %q: %v", data.Id(), checkErr)
+	}
+
 	return nil
 }
 
+
+
+func resourceCastaiClusterUpdate(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*ProviderConfig).api
+
+	log.Printf("[DEBUG] Cluster %q autoscaling policies update", data.Id())
+	autoscalerParams, ok := data.Get(PolicyFieldAutoscalerPolicies).([]interface{})
+	if !ok || autoscalerParams == nil || len(autoscalerParams) == 0 || autoscalerParams[0] == nil {
```

Comment:
isn't `!ok || len(autoscalerParams) == 0` sufficient?

Reply:
I stole from somewhere this, I checked that autoscalerParams == nil is redundant as nil will be true for len(autoscalerParams)==0, removed. Not sure regarding last one, leaving just in case if list is empty